### ISSUE TITLE
segfault at 2d700000232 ip 000002d700000232 sp 00007ffe9a444a28 error 14

### ISSUE THREAD

**Issue by fenidik** [vacuum-im/vacuum-im#898]
Vacuum fails with core dumps (can't find dump file anyway :) )

Distributor ID: Ubuntu
Description:    Ubuntu 16.04 LTS
Release:    16.04
Codename:   xenial

ii  libvacuumutils37                            1.3.0+git201605121411-xenial1                       amd64        Full-featured crossplatform Jabber/XMPP client (shared library)
ii  vacuum                                      1.3.0+git201605121411-xenial1                       amd64        Full-featured crossplatform Jabber/XMPP client

(gdb) r
Starting program: /usr/bin/vacuum 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7fffe852e700 (LWP 24338)]
[New Thread 0x7fffe76e7700 (LWP 24339)]
[New Thread 0x7fffe6391700 (LWP 24340)]
[New Thread 0x7fffdffff700 (LWP 24341)]
[New Thread 0x7fffdf7fe700 (LWP 24342)]
[Thread 0x7fffdffff700 (LWP 24341) exited]
[New Thread 0x7fffdffff700 (LWP 24344)]
[New Thread 0x7fffbb1cb700 (LWP 24345)]
[New Thread 0x7fffba9ca700 (LWP 24346)]
[New Thread 0x7fffba1c9700 (LWP 24347)]
[New Thread 0x7fffb97a9700 (LWP 24348)]
[New Thread 0x7fffb8d9a700 (LWP 24350)]
[New Thread 0x7fffa3fff700 (LWP 24352)]
[New Thread 0x7fffa37fe700 (LWP 24353)]
[Thread 0x7fffb8d9a700 (LWP 24350) exited]
[Thread 0x7fffdf7fe700 (LWP 24342) exited]
[Thread 0x7fffa3fff700 (LWP 24352) exited]
[Thread 0x7fffa37fe700 (LWP 24353) exited]
[Thread 0x7fffdffff700 (LWP 24344) exited]
[Thread 0x7fffbb1cb700 (LWP 24345) exited]
[Thread 0x7fffba1c9700 (LWP 24347) exited]
[Thread 0x7fffba9ca700 (LWP 24346) exited]
[New Thread 0x7fffba9ca700 (LWP 24444)]
[Thread 0x7fffba9ca700 (LWP 24444) exited]

Thread 1 "vacuum" received signal SIGSEGV, Segmentation fault.
0x00007fffc174ab9e in ReceiversWidget::onRosterItemReceived(IRoster*, IRosterItem const&, IRosterItem const&) () from /usr/lib/vacuum/plugins/libmessagewidgets.so

**Comment by krab** [comment]
> (gdb) r

bt?

**Comment by Lion34** [comment]
А логи от падения не остались в ~/.vacuum/logs?

**Comment by fenidik** [comment]
```
(gdb) r
Starting program: /usr/bin/vacuum 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7fffe852e700 (LWP 26416)]
[New Thread 0x7fffe7d2d700 (LWP 26417)]
[New Thread 0x7fffe752c700 (LWP 26418)]
[New Thread 0x7fffdffff700 (LWP 26419)]
[New Thread 0x7fffdf7fe700 (LWP 26420)]
[Thread 0x7fffdffff700 (LWP 26419) exited]
[New Thread 0x7fffdffff700 (LWP 26423)]
[New Thread 0x7fffbb1cb700 (LWP 26424)]
[New Thread 0x7fffba9ca700 (LWP 26425)]
[New Thread 0x7fffba1c9700 (LWP 26426)]
[New Thread 0x7fffb97a9700 (LWP 26429)]
[New Thread 0x7fffb8d9a700 (LWP 26431)]
[New Thread 0x7fffa3fff700 (LWP 26433)]
[New Thread 0x7fffa37fe700 (LWP 26434)]
[New Thread 0x7fffa2bf3700 (LWP 26439)]
[New Thread 0x7fffa1d7d700 (LWP 26440)]
[New Thread 0x7fffa10e1700 (LWP 26441)]
[Thread 0x7fffb8d9a700 (LWP 26431) exited]
[New Thread 0x7fffb8d9a700 (LWP 26443)]
[Thread 0x7fffdf7fe700 (LWP 26420) exited]

Thread 1 "vacuum" received signal SIGSEGV, Segmentation fault.
0x0000000000000000 in ?? ()
(gdb) bt
#0  0x0000000000000000 in ?? ()
#1  0x00007fffc174abb2 in ReceiversWidget::onRosterItemReceived(IRoster*, IRosterItem const&, IRosterItem const&) () from /usr/lib/vacuum/plugins/libmessagewidgets.so
#2  0x00007fffc177e3d7 in ?? () from /usr/lib/vacuum/plugins/libmessagewidgets.so
#3  0x00007ffff6868010 in QMetaObject::activate(QObject*, QMetaObject const*, int, void**) () from /usr/lib/x86_64-linux-gnu/libQtCore.so.4
#4  0x00007fffbfba4a7c in RosterManager::rosterItemReceived(IRoster*, IRosterItem const&, IRosterItem const&) () from /usr/lib/vacuum/plugins/libroster.so
#5  0x00007fffbfba4eed in ?? () from /usr/lib/vacuum/plugins/libroster.so
#6  0x00007ffff6868010 in QMetaObject::activate(QObject*, QMetaObject const*, int, void**) () from /usr/lib/x86_64-linux-gnu/libQtCore.so.4
#7  0x00007fffbfba3eda in Roster::itemReceived(IRosterItem const&, IRosterItem const&) () from /usr/lib/vacuum/plugins/libroster.so
#8  0x00007fffbfb95e2d in Roster::processItemsElement(QDomElement const&, bool) () from /usr/lib/vacuum/plugins/libroster.so
#9  0x00007fffbfb9863b in Roster::stanzaReadWrite(int, Jid const&, Stanza&, bool&) () from /usr/lib/vacuum/plugins/libroster.so
#10 0x00007fffbdad4166 in StanzaProcessor::processStanza(Jid const&, Stanza&, int) const () from /usr/lib/vacuum/plugins/libstanzaprocessor.so
#11 0x00007fffbdad50c6 in StanzaProcessor::xmppStanzaIn(IXmppStream*, Stanza&, int) () from /usr/lib/vacuum/plugins/libstanzaprocessor.so
#12 0x00007fffbc9d6b35 in XmppStream::processStanzaHandlers(Stanza&, bool) () from /usr/lib/vacuum/plugins/libxmppstreams.so
#13 0x00007fffbc9d8821 in XmppStream::onParserElement(QDomElement const&) () from /usr/lib/vacuum/plugins/libxmppstreams.so
#14 0x00007fffbc9de099 in ?? () from /usr/lib/vacuum/plugins/libxmppstreams.so
#15 0x00007ffff6868010 in QMetaObject::activate(QObject*, QMetaObject const*, int, void**) () from /usr/lib/x86_64-linux-gnu/libQtCore.so.4
#16 0x00007fffbc9dc4f5 in StreamParser::element(QDomElement const&) () from /usr/lib/vacuum/plugins/libxmppstreams.so
#17 0x00007fffbc9db71d in StreamParser::parseData(QByteArray const&) () from /usr/lib/vacuum/plugins/libxmppstreams.so
#18 0x00007fffbc9d58ce in XmppStream::onConnectionReadyRead(long long) () from /usr/lib/vacuum/plugins/libxmppstreams.so
#19 0x00007fffbc9de0dc in ?? () from /usr/lib/vacuum/plugins/libxmppstreams.so
#20 0x00007ffff6868010 in QMetaObject::activate(QObject*, QMetaObject const*, int, void**) () from /usr/lib/x86_64-linux-gnu/libQtCore.so.4
#21 0x00007fffc6a69f62 in DefaultConnection::readyRead(long long) () from /usr/lib/vacuum/plugins/libdefaultconnection.so
#22 0x00007fffc6a6a1fd in ?? () from /usr/lib/vacuum/plugins/libdefaultconnection.so
#23 0x00007ffff6868010 in QMetaObject::activate(QObject*, QMetaObject const*, int, void**) () from /usr/lib/x86_64-linux-gnu/libQtCore.so.4
#24 0x00007ffff5b08ed4 in ?? () from /usr/lib/x86_64-linux-gnu/libQtNetwork.so.4
#25 0x00007ffff5b02071 in ?? () from /usr/lib/x86_64-linux-gnu/libQtNetwork.so.4
#26 0x00007ffff6868010 in QMetaObject::activate(QObject*, QMetaObject const*, int, void**) () from /usr/lib/x86_64-linux-gnu/libQtCore.so.4
#27 0x00007ffff5ae342e in ?? () from /usr/lib/x86_64-linux-gnu/libQtNetwork.so.4
#28 0x00007ffff5aec2dd in ?? () from /usr/lib/x86_64-linux-gnu/libQtNetwork.so.4
#29 0x00007ffff6fc6fdc in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /usr/lib/x86_64-linux-gnu/libQtGui.so.4
#30 0x00007ffff6fcdf16 in QApplication::notify(QObject*, QEvent*) () from /usr/lib/x86_64-linux-gnu/libQtGui.so.4
#31 0x00007ffff685390d in QCoreApplication::notifyInternal(QObject*, QEvent*) () from /usr/lib/x86_64-linux-gnu/libQtCore.so.4
#32 0x00007ffff68848a2 in ?? () from /usr/lib/x86_64-linux-gnu/libQtCore.so.4
#33 0x00007ffff4962127 in g_main_context_dispatch () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#34 0x00007ffff4962380 in ?? () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#35 0x00007ffff496242c in g_main_context_iteration () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#36 0x00007ffff68842ae in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib/x86_64-linux-gnu/libQtCore.so.4
#37 0x00007ffff7074616 in ?? () from /usr/lib/x86_64-linux-gnu/libQtGui.so.4
#38 0x00007ffff685218f in QEventLoop::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib/x86_64-linux-gnu/libQtCore.so.4
#39 0x00007ffff68524f5 in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib/x86_64-linux-gnu/libQtCore.so.4
#40 0x00007ffff68584b9 in QCoreApplication::exec() () from /usr/lib/x86_64-linux-gnu/libQtCore.so.4
#41 0x000000000040ed98 in main ()

```

**Comment by fenidik** [comment]
last messages in log:

```

09:20:23.811    +141     INF    [VCardManager] [XX@XX] User vCard loaded, jid=YY5@YY, id=sid_57
09:20:23.849    +38  INF    [VCardManager] [XX@XX] User vCard loaded, jid=YY6@YY, id=sid_62
09:20:23.852    +3   INF    [VCardManager] [XX@XX] User vCard loaded, jid=YY7@YY, id=sid_59
09:20:23.859    +7   INF    [VCardManager] [XX@XX] User vCard loaded, jid=YY1@YY, id=sid_58
09:20:25.383    +1524    INF    [VCardManager] [XX@XX] User vCard load request sent to=YY2@YY, id=sid_63
09:20:25.446    +63 *WNG    [VCardManager] [XX@XX] Failed to load user vCard, jid=YY2@YY, id=sid_63: item-not-found
09:20:30.390    +4944    INF    [VCardManager] [XX@XX] User vCard load request sent to=YY3@YY, id=sid_68
09:20:30.453    +63  INF    [VCardManager] [XX@XX] User vCard loaded, jid=YY3@YY, id=sid_68

```

**Comment by Lion34** [comment]
А при каком действии с клиентом падает?

**Comment by fenidik** [comment]
Просто падает. Даже когда окно неактивно. Когда активно тоже падает :) Не могу поймать закономерность.

**Comment by Lion34** [comment]
Судя по логам приходит ростер-пуш, может есть предположение, что это может быть? У меня никак не получается проявить.

**Comment by Lion34** [comment]
Fixed in d660dafacb072687f6797338f2b07a7dd879694c
